core/org.wso2.carbon.base/src/test/java/org/wso2/carbon/base/CarbonBaseUtilsTest.java

```diff
@@ -0,0 +1,178 @@
+            resourcePath = resourcePath + "policy-test.policy";
+            System.setProperty("java.security.policy", resourcePath);
+            System.setSecurityManager(new SecurityManager());
+        }
+    }
+
+    @AfterClass
+    public void clearSecurityManager() {
+        System.clearProperty("java.security.policy");
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testGetCarbonConfigDirPath() throws ClassNotFoundException, IllegalAccessException,
+            NoSuchFieldException {
+        // Setting env variables for testing getCarbonConfigDirPath() in CarbonBaseUtils
+        setEnvironmentVariables(CarbonBaseConstants.CARBON_CONFIG_DIR_PATH_ENV, "/wso2am-2.1.0/repository/conf");
+        assertEquals(CarbonBaseUtils.getCarbonConfigDirPath(),
+                "/wso2am-2.1.0/repository/conf", "Must provide the location where carbon.xml resides");
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testGetServerXML() {
+        System.setProperty(CarbonBaseConstants.CARBON_CONFIG_DIR_PATH, "/wso2am-2.1.0/repository/conf");
+        String absoluteLocation = CarbonBaseUtils.getServerXml();
+        assertEquals(absoluteLocation, "/wso2am-2.1.0/repository/conf/carbon" + ".xml", "Must match the formed " +
+                "absoluteLocation of carbon.xml");
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testGetCarbonHome() {
+        System.setProperty(CarbonBaseConstants.CARBON_HOME, "/wso2am-2.1.0");
+        assertEquals(CarbonBaseUtils.getCarbonHome(), "/wso2am-2.1.0", "Must provide the carbon home ");
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testCheckSecurityWithClasses() {
+        List<String> allowedClasses = Arrays.asList("org.junit.runners.model.FrameworkMethod$1.runReflectiveCall",
+                "org.wso2.carbon.base.CarbonBaseUtils.checkSecurity", "org.junit.internal.runners.statements" +
+                        ".InvokeMethod.evaluate", "org.junit.runner.JUnitCore.run", "org.junit.runners.ParentRunner.run");
+        CarbonBaseUtils.checkSecurity(allowedClasses);
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testCheckSecurityWithMethods() {
+        Map<String, String> allowedMethods = new HashMap<>();
+        allowedMethods.put("sun.reflect.NativeMethodAccessorImpl", "invoke");
+        allowedMethods.put("sun.reflect.DelegatingMethodAccessorImpl", "invoke");
+        allowedMethods.put("java.lang.reflect.Method", "invoke");
+        allowedMethods.put("org.junit.internal.runners.statements.InvokeMethod", "evaluate");
+        CarbonBaseUtils.checkSecurity(allowedMethods);
+    }
+
+    /**
+     * Set environment variable for a given key and value.
+     *
+     * @param key   Environment variable key.
+     * @param value Environment variable value.
+     */
+    public static void setEnvironmentVariables(String key, String value) throws IllegalAccessException,
```

Comment:
why this is public static? who uses this? if this is within this class, then shouldn't it be private?